tests/test_pex_binary.py

```diff
@@ -88,6 +90,17 @@ def test_clp_constraints_txt():
   assert options.constraint_files == ['requirements1.txt']
 
 
+def test_clp_preamble_file():
+  with NamedTemporaryFile() as tmpfile:
+    tmpfile.write(to_bytes('print "foo!"'))
+    tmpfile.flush()
+
+    # XXX ? can we assert that the PEXBuilder actually gets the contents of the preamble file?
```

Comment:
Yeah, how about via execution of a built pex with pre-amble? If the preamble echoed, say and exited with a known exit code you could test bout the stdout and the exit code are as expected.